testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/log/InvalidTransactionAttributeTestCase.java

```diff
@@ -0,0 +1,55 @@
+package org.jboss.as.test.integration.ejb.log;
+
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+
+/**
+ * Tests if the WARN message was logged when default transaction attribute was used on SFSB lifecyle method
+ * Test for [ WFLY-9509 ].
+ *
+ * @author Daniel Cihak
+ */
+@RunWith(Arquillian.class)
+public class InvalidTransactionAttributeTestCase {
+
+    private static final String ARCHIVE_NAME = "InvalidTransactionAttribute";
+
+    @ArquillianResource
+    Deployer deployer;
+
+    @Deployment(name = "invalidtransactionattribute", testable = false, managed = false)
+    public static Archive<?> deploy() {
+        JavaArchive jar = ShrinkWrap.create(JavaArchive.class, ARCHIVE_NAME + ".jar");
+        jar.addClasses(InvalidTransactionAttributeTestCase.class, StatefulBean.class, StatefulInterface.class);
+        jar.addAsManifestResource(InvalidTransactionAttributeTestCase.class.getPackage(), "ejb-jar.xml", "ejb-jar.xml");
+        return jar;
+    }
+
+    @Test
+    public void testInvalidTransactionAttributeWarnLogged() throws Exception {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream oldOut = System.out;
+        System.setOut(new PrintStream(baos));
+
+        deployer.deploy("invalidtransactionattribute");
+
+        System.setOut(oldOut);
```

Comment:
This must be done in a finally block.

Reply:
There is no exception to be thrown in the test method. I removed "throws Exception" in my last commit.

Reply:
The absence of a possible checked exception does not mean code will not fail. RuntimeException, Error (including AssertionError), new code added later, these all can result in necessary cleanup not occurring.